x-pack/plugins/uptime/server/lib/requests/get_monitor_states.ts

```diff
@@ -68,11 +69,113 @@ export const getMonitorStates: UMElasticsearchQueryFn<
     statusFilter
   );
 
-  const page = await fetchPage(queryContext);
+  const size = Math.min(queryContext.size, QUERY.DEFAULT_AGGS_CAP);
+
+  const iterator = new MonitorSummaryIterator(queryContext);
+  const page = await iterator.nextPage(size);
+
+  const histograms = await getHistogramForMonitors(
+    queryContext,
+    page.monitorSummaries.map((s) => s.monitor_id)
+  );
+
+  page.monitorSummaries.forEach((s) => {
+    s.histogram = histograms[s.monitor_id];
+  });
```

Comment:
if we remove this part, i have noticed monitor list query against https://dev-next-oblt.elastic.dev/ is at-least 1 second faster.

Reply:
perhaps in future we can move histograms to a separate query.

Reply:
GetMonitorStatesParams type and GetMonitorStatesParams and FetchMonitorStatesQueryArgs are duplicates of each other. one should be removed.

Reply:
WRT the type dupe, I'm OK the duplication since one is for the backend and the other is for the frontend. While they are the same now, they are semantically different. It's more defensive code-wise to separate parameterization that is used when accepting user-facing requests from backend code. While it's not the most likely vector, it's OK IMHO.